examples/gallery/histograms/scatter_and_histograms.py

```diff
@@ -2,70 +2,74 @@
 Scatter plot with histograms
 ============================
 
-To create a scatter plot with histograms at the sides of the plot one
-can use :meth:`pygmt.Figure.plot` in combination with
-:meth:`pygmt.Figure.histogram`. The positions of the histograms are plotted
-by offsetting them from the main scatter plot figure using
-:meth:`pygmt.Figure.shift_origin`.
+To create a scatter plot with histograms at the sides of the plot one can use
+:meth:`pygmt.Figure.plot` in combination with :meth:`pygmt.Figure.histogram`.
+The positions of the histograms are plotted by offsetting them from the main scatter
+plot using :meth:`pygmt.Figure.shift_origin`.
 """
 
 # %%
 import numpy as np
 import pygmt
 
-# Generate random data from a standard normal distribution centered on 0
-# with a standard deviation of 1
-rng = np.random.default_rng(seed=19680801)
+# Generate random x, y coordinates from a standard normal distribution.
+# x are centered on 0 with a standard deviation of 1, and y are centered on 30 with a
+# standard deviation of 2.
+rng = np.random.default_rng()
 x = rng.normal(loc=0, scale=1, size=1000)
-y = rng.normal(loc=0, scale=1, size=1000)
+y = rng.normal(loc=30, scale=2, size=1000)
 
-# Get axis limits
-xymax = max(np.max(np.abs(x)), np.max(np.abs(y)))
+# Get axis limits from the data limits. Extend the limits by 0.5 to add some margin.
+xmin = np.floor(x.min()) - 0.5
+xmax = np.ceil(x.max()) + 0.5
+ymin = np.floor(y.min()) - 0.5
+ymax = np.ceil(y.max()) + 0.5
 
+# Set fill color for symbols and bars
+fill = "seagreen"
+
+# Set the dimensions of the scatter plot.
+width, height = 10, 8
 
 fig = pygmt.Figure()
 fig.basemap(
-    region=[-xymax - 0.5, xymax + 0.5, -xymax - 0.5, xymax + 0.5],
-    projection="X10c/10c",
-    frame=["WSrt", "a1"],
+    region=[xmin, xmax, ymin, ymax],
+    projection=f"X{width}/{height}",
```

Comment:
I'm more inclined to skip the unit if possible. For projections, it's fine to keep the unit or not, but for other parameters, e.g, `xshift`, skipping the unit makes more sense (i.e., `xshift=width + 2`, rather than `xshift=f"{width + 2}c`.